### DISCUSSION TITLE
Data at a Glance view

### DISCUSSION THREAD

**Discussion by jqnatividad** [ckan/ckan#6005] [Ideas]
A quick snapshot of a package showing:
 1. utilization metadata 
    - a top-level "Google Analytics" like digest
    - downloads
    - api calls
    - top referrers
    - data transfer bandwidth
    - top Solr/datatables searches for the package
 2. descriptive statistics for each column (e.g. top n values, min/max, frequencies, etc)
 3.  storage metadata (number of rows, size, etc.)
 4.  related items
     * automatic relationships
         - other packages based on similarity of data dictionaries
         - other packages based on similarity of descriptive statistics
         - external sites based on embeds of resource views
         - other packages as gleaned from user traffic
     * curated relationships
         - as entered by the data publisher (e.g. apps, news items  using the package)

This should help all stakeholders get better visibility into how the data is used.

**Comment by wardi** [comment]
Love this idea. Will need background jobs to compute some of the more expensive metrics. Are you thinking of starting with an extension or targeting core?

**Comment by jqnatividad** [comment]
Was thinking of starting with an extension, but it will need to be more of an app or "solution bundle" composed of different components.

As you pointed out, several metrics will need to be precomputed and was thinking of storing them as "system resources" (https://github.com/ckan/ideas/issues/260), with the background jobs being run with Airflow using [aircan](https://github.com/datopian/aircan) perhaps?

Also was thinking of using [Grafana](https://github.com/grafana/grafana)/[Prometheus](https://prometheus.io/docs/introduction/overview/) as they already have connectors to nginx, solr and postgres that can be used to build out several elements of the "At a Glance" dashboard.

With the "At a Glance" extension being primarily a wrapper around the Grafana dashboard, extending elements of the package template and/or as a "system resource view."

It will be a much improved version of Socrata's data synopsis page (e.g. https://data.cityofnewyork.us/Social-Services/311-Service-Requests-from-2010-to-Present/erm2-nwe9) built with best-of-breed open source components.

Of course, the "At a Glance" dashboard is but one bundled configurable Grafana dashboard. CKAN admins can build out their own dashboards for other purposes.

What do you think?
